### ISSUE TITLE
Error if I have uploaded files and try to upload new files

### ISSUE THREAD

**Issue by mohithg** [pqina/react-filepond#3]
Hello @pqina,
I am using react-filepond and found this issue

```
filepond.js:6527 Uncaught TypeError: Cannot read property 'filename' of null
    at Object.itemProcessed [as DID_COMPLETE_ITEM_PROCESSING]
```
This happens when I already have some files to display and try to upload new files with my custom process function
eg:
```
<FilePond
          allowMultiple={true}
          server={{
            process: this.handleProcessing.bind(this),
          }}
        >
          <File key={0} source={'index.html'} type="local" />
        </FilePond>
```
Now If i try to upload new files, the files even though gets processed, will show this error in console and blob gets hidden.

**Comment by mohithg** [comment]
One possible fix I found is 
```
this._pond.setOptions({
      files,
    });
```
set this inside `componentDidMount()` and remove setting of files in `componentDidUpdate()`

**Comment by mohithg** [comment]
Fixed in https://github.com/pqina/react-filepond/pull/4 if you like to merge

**Comment by rikschennink** [comment]
I will run some tests to confirm this. Removing the `setOptions` call from `componentDidUpdate` will prevent updating of the component, so that's not a solution.

**Comment by rikschennink** [comment]
@mohithg Can you post your processing method?

**Comment by mohithg** [comment]
```async handleProcessing(fieldName, file, metadata, load, error, progress, abort) {
    await upload(file);
    load(file)
  }
```
@rikschennink you can remove updating the files in the `componentDidUpdate`, because filepond will have files when dropped, so we need not update it always right expect for initial loading if there are any uploaded files?

**Comment by rikschennink** [comment]
@mohithg Okay, that looks fine. Because of how your initial issue was formulated it sounded like you were adding files to FilePond from your process method, just wanted to be sure :-)

I will try to reproduce this problem this afternoon or Thursday. Want to make sure the fix is suitable before merging.

**Comment by parweb** [comment]
Any news on this ?

**Comment by rikschennink** [comment]
@parweb @mohithg My apologies, I totally lost track of this.

I think I reproduced the issue by setting a files array, then adding a file manually, then updating a prop to trigger the `componentDidUpdate` causing the files list to be reset. I've made a small change to compare the previous files array with the new one so it's still possible to update the files using the `<File/>` component.

I've just uploaded and published fix version `2.0.4`
